src/signals/settings/__tests__/index.test.js

```diff
@@ -23,6 +24,41 @@ describe('signals/settings', () => {
     reactRouterDom.useLocation.mockRestore();
   });
 
+  it('should have props from action creator', () => {
+    const tree = mount(withAppContext(<SettingsModule />));
+
+    const containerProps = tree.find(Module).props();
+
+    expect(containerProps.onFetchDepartments).not.toBeUndefined();
+    expect(typeof containerProps.onFetchDepartments).toEqual('function');
+
+    expect(containerProps.onFetchPermissions).not.toBeUndefined();
+    expect(typeof containerProps.onFetchPermissions).toEqual('function');
+
+    expect(containerProps.onFetchRoles).not.toBeUndefined();
+    expect(typeof containerProps.onFetchRoles).toEqual('function');
+  });
+
+  it('should initiate fetches on mount', () => {
```

Comment:
Performing async tasks here. `async ()`

Reply:
Not necessarily; the actual fetch is async, but not the calls that initiate the fetch and that is what this particular test is about. The fetch itself is not the responsibility of the `settings` module, so that is why I didn't test it here.